src/participant/participant.ts

```diff
@@ -333,40 +261,47 @@ export default class ParticipantController {
     }
   }
 
-  async _selectDatabaseWithCommandPalette(): Promise<string | undefined> {
+  async _selectDatabaseWithQuickPick(): Promise<string | undefined> {
     const databases = await this.getDatabaseQuickPicks();
     const selectedQuickPickItem = await vscode.window.showQuickPick(databases, {
       placeHolder: 'Select a database...',
     });
     return selectedQuickPickItem?.data;
   }
 
-  async selectDatabaseWithParticipant(name: string): Promise<boolean> {
-    if (!name) {
-      const selectedName = await this._selectDatabaseWithCommandPalette();
-      this._setDatabaseName(selectedName);
-    } else {
-      this._setDatabaseName(name);
+  async selectDatabaseWithParticipant({
+    databaseName: _databaseName,
+  }: {
+    databaseName: string;
+  }): Promise<boolean> {
+    let databaseName: string | undefined = _databaseName;
+    if (!databaseName) {
+      databaseName = await this._selectDatabaseWithQuickPick();
+      if (!databaseName) {
+        return false;
+      }
     }
 
     return vscode.commands.executeCommand('workbench.action.chat.open', {
-      query: `@MongoDB /query ${this._databaseName || ''}`,
+      // Add a database name message to the chat so future messages know which namespace to use.
+      query: `@MongoDB /query ${databaseNameInCommandResponseKey}${databaseName}`,
```

Comment:
Can we avoid these databaseNameInCommandResponseKey and collectionNameInCommandResponseKey prefixes? We don't use one for the connection name but use it for database and collection names. Can we maybe attach those to the prompt when we iterate over history?

Reply:
We could also return some message with the generated query if we have changed a namespace. Something like `Here’s an example generated for the 'database.collection' namespace:`, that can be attached to further prompt from the history.

Reply:
Removed the prefixes 👍